Game/Source/GDKTestGyms/Private/CounterComponent.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "CounterComponent.h"
+
+#include "Engine/World.h"
+#include "Kismet/GameplayStatics.h"
+
+DEFINE_LOG_CATEGORY(LogCounterComponent);
+
+UCounterComponent::UCounterComponent()
+	: Timer(0.0f)
+{
+	PrimaryComponentTick.bCanEverTick = true;
+}
+
+void UCounterComponent::TickComponent(float DeltaTime, ELevelTick TickType, FActorComponentTickFunction* ThisTickFunction)
+{
+	Super::TickComponent(DeltaTime, TickType, ThisTickFunction);
+
+	Timer -= DeltaTime;
+	if (Timer <= 0.0f)
+	{
+		Timer = ReportFrequency;
+		UpdateCachedActorCounts();
+	}
+}
+
+int32 UCounterComponent::GetActorClassCount(TSubclassOf<AActor> ActorClass) const
```

Comment:
is this function even used? wouldn't we just report the current situation in update? as in - do we need the caching?

Reply:
You're right this isn't currently used, but this is just reverting files that already existed, so will leave for the time being.